File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/SedStatus.java
package no.nav.melosys.eessi.controller.dto;

import java.util.Arrays;
import lombok.Getter;
import org.springframework.util.StringUtils;

@Getter
public enum SedStatus {
    SENDT("SENDT", "SENT"),
    UTKAST("UTKAST", "NEW"),
    TOM("TOM", "EMPTY"),
    MOTTATT("MOTTATT", "RECEIVED"),
    AVBRUTT("AVBRUTT", "CANCELLED");

    private final String norskStatus;

    private final String engelskStatus;

    SedStatus(String norskStatus, String engelskStatus) {
        this.engelskStatus = engelskStatus;
        this.norskStatus = norskStatus;
    }

    public static SedStatus fraNorskStatus(String norskStatus) {

        if (StringUtils.isEmpty(norskStatus)) {
            return null;
        }

        return valueOf(norskStatus.toUpperCase());
    }

    public static SedStatus fraEngelskStatus(String engelskStatus) {

        if (StringUtils.isEmpty(engelskStatus)) {
            return null;
        }

        return Arrays.stream(values())
                .filter(status -> engelskStatus.equalsIgnoreCase(status.engelskStatus))
                .findFirst()
                .orElse(null);
    }

    public static boolean erGyldigEngelskStatus(String engelskstatus) {
        return MOTTATT.engelskStatus.equalsIgnoreCase(engelskstatus)
                || SENDT.engelskStatus.equalsIgnoreCase(engelskstatus);
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/SedStatus.kt
package no.nav.melosys.eessi.controller.dto

enum class SedStatus(val norskStatus: String, val engelskStatus: String) {
    SENDT("SENDT", "SENT"),
    UTKAST("UTKAST", "NEW"),
    TOM("TOM", "EMPTY"),
    MOTTATT("MOTTATT", "RECEIVED"),
    AVBRUTT("AVBRUTT", "CANCELLED");

    companion object {
        @JvmStatic
        fun fraNorskStatus(norskStatus: String?): SedStatus? {
            return if (norskStatus.isNullOrBlank()) null
            else valueOf(norskStatus.uppercase())
        }

        @JvmStatic
        fun fraEngelskStatus(engelskStatus: String?): SedStatus? {
            return if (engelskStatus.isNullOrBlank()) null
            else entries.firstOrNull { it.engelskStatus.equals(engelskStatus, ignoreCase = true) }
        }

        @JvmStatic
        fun erGyldigEngelskStatus(engelskStatus: String?): Boolean {
            return engelskStatus.equals(MOTTATT.engelskStatus, ignoreCase = true) ||
                engelskStatus.equals(SENDT.engelskStatus, ignoreCase = true)
        }
    }
}
